common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -293,6 +296,28 @@ public void initChannel(SocketChannel ch) {
     } else if (cf.cause() != null) {
       throw new IOException(String.format("Failed to connect to %s", address), cf.cause());
     }
+    if (context.sslEncryptionEnabled()) {
+      final SslHandler sslHandler = cf.channel().pipeline().get(SslHandler.class);
+      Future<Channel> future = sslHandler.handshakeFuture().addListener(
+        new GenericFutureListener<Future<Channel>>() {
+          @Override
+          public void operationComplete(final Future<Channel> handshakeFuture) {
+            if (handshakeFuture.isSuccess()) {
+              logger.debug("{} successfully completed TLS handshake to ", address);
+            } else {
+              logger.info(
+                "failed to complete TLS handshake to " + address, handshakeFuture.cause());
+              cf.channel().close();
+            }
+          }
+      });
+      if (!future.await(conf.connectionTimeoutMs())) {
+        logger.info("failed to connect to " + address + " within connection timeout");
```

Comment:
QQ: Will we have two log messages for the same failure ? Here and in `operationComplete` ?

Reply:
I believe so - will remove this.